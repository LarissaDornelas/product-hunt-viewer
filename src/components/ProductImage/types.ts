export enum ImageSize {
  SMALL = "15px",
  MEDIUM = "40px",
  LARGE = "80px",
}

export interface IProductImage {
  imageUrl: string;
  size: ImageSize;
}
