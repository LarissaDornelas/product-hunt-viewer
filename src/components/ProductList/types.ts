export interface IProductItem {
  id: string;
  name: string;
  description: string;
  votesCount: string;
  thumbnail: string;
}

export interface IProductList {
  data: IProductItem[];
  loading: boolean;
  handleFetchMore: () => void;
}
