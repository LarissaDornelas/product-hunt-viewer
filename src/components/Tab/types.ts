import { Order } from "hooks/posts/types";

export interface ITabItem {
  title: string;
  active: boolean;
  value: Order;
}
export interface ITab {
  tabItems: ITabItem[];
  toggleTab: (index: number) => void;
}
