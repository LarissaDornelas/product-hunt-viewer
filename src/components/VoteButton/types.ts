export interface IVoteButton {
  votes: number;
}
