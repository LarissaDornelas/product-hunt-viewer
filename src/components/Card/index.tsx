import React from "react";

import { Container } from "./styles";

const Card: React.FC = (props) => {
  const { children } = props;
  return <Container data-testid="card-container">{children}</Container>;
};

export { Card };
