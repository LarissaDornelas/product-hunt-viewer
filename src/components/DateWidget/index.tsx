import React from "react";
import { format } from "date-fns";

import { Container } from "./styles";

const DateWidget: React.FC = () => {
  return (
    <Container data-testid="date-widget-container">
      <p>{format(new Date(), "'Today', Lo LLL ")}</p>
    </Container>
  );
};

export { DateWidget };
