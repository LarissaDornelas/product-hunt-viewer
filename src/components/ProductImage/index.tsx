import React from "react";

import { Container } from "./styles";
import { IProductImage } from "./types";

const ProductImage: React.FC<IProductImage> = (props) => {
  const { imageUrl, size } = props;

  return (
    <Container
      data-testid="product-image-container"
      imageUrl={imageUrl}
      size={size}
    />
  );
};

export { ProductImage };
