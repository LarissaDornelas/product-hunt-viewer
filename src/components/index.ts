import { ProfileImage } from "./ProfileImage";
import { DateWidget } from "./DateWidget";
import { SearchButton } from "./SearchButton";
import { Tab } from "./Tab";
import { Card } from "./Card";

export { ProfileImage, DateWidget, SearchButton, Tab, Card };
