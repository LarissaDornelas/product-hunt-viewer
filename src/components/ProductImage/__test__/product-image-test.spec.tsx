import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

import { ProductImage } from "components";
import { ImageSize } from "../types";

describe("Component: ProductImage", () => {
  afterEach(cleanup);
  const imageUrl = "http://test.com";

  it("Should render without crashing", () => {
    const container = document.createElement("div");

    ReactDOM.render(
      <ProductImage imageUrl={imageUrl} size={ImageSize.LARGE} />,
      container
    );
  });

  it("Should render ProductImage correctly", () => {
    const { getByTestId } = render(
      <ProductImage imageUrl={imageUrl} size={ImageSize.LARGE} />
    );

    expect(getByTestId("product-image-container")).toHaveStyle({
      backgroundImage: "url(http://test.com)",
      backgroundRepeat: "no-repeat",
      backgroundSize: "cover",
      height: "80px",
      width: "80px",
      borderRadius: "10px",
    });
  });
  it("matches snapshot", () => {
    const tree = renderer
      .create(<ProductImage imageUrl={imageUrl} size={ImageSize.LARGE} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
