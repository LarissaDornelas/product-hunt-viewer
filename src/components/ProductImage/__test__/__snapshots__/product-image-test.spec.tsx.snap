// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProductImage matches snapshot 1`] = `
<div
  className="sc-fubCfw hbzESy"
  data-testid="product-image-container"
  size="80px"
/>
`;
